src/file_organizer/services/deduplication/embedder.py

```
@@ -124,15 +126,17 @@ def transform(self, document: str) -> np.ndarray:
 
         # Check cache
         doc_hash = self._hash_document(document)
-        if doc_hash in self.embedding_cache:
-            logger.debug(f"Cache hit for document (hash={doc_hash[:8]})")
-            return self.embedding_cache[doc_hash]
+        with self._cache_lock:
+            if doc_hash in self.embedding_cache:
+                logger.debug("Cache hit for document (hash={})", doc_hash[:8])
```

Ruff (0.15.6)

[error] Too many arguments for `logging` format string

(PLE1205)